guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,750 @@
+KerasCV has a predefined specificication for bounding boxes.
+To comply with this, you
+should package your bounding boxes into a dictionary matching the
+specification below:
+
+```
+bounding_boxes = {
+    # num_boxes may be a Ragged dimension
+    'boxes': Tensor(shape=[batch, num_boxes, 4]),
+    'classes': Tensor(shape=[batch, num_boxes])
+}
+```
+
+`bounding_boxes['boxes']` contains the coordinates of your bounding box in a KerasCV
+supported `bounding_box_format`.
+KerasCV requires a `bounding_box_format` argument in all components that process
+bounding boxes.
+This is done to maximize your ability to plug and play individual components
+into their object detection pipelines, as well as to make code self-documenting
+across object detection pipelines.
+
+To match the KerasCV API style, it is recommended that when writing a
+custom data loader, you also support a `bounding_box_format` argument.
+This makes it clear to those invoking your data loader what format the bounding boxes
+are in.
+In this example, we format our boxes to `xywh` format.
+
+For example:
+
+```python
+train_ds, ds_info = your_data_loader.load(
+    split='train', bounding_box_format='xywh', batch_size=8
+)
+```
+
+Clearly yields bounding boxes in the format `xywh`.  You can read more about
+KerasCV bounding box formats [in the API docs](https://keras.io/api/keras_cv/bounding_box/formats/).
+
+Our data comesloaded into the format
+`{"images": images, "bounding_boxes": bounding_boxes}`.  This format is
+supported in all KerasCV preprocessing components.
+
+Let's load some data and verify that our data looks as we expect it to.
+"""
+
+
+def unpackage_raw_tfds_inputs(inputs, bounding_box_format):
+    image = inputs["image"]
+    boxes = keras_cv.bounding_box.convert_format(
+        inputs["objects"]["bbox"],
+        images=image,
+        source="rel_yxyx",
+        target=bounding_box_format,
+    )
+    bounding_boxes = {
+        "classes": tf.cast(inputs["objects"]["label"], dtype=tf.float32),
+        "boxes": tf.cast(boxes, dtype=tf.float32),
+    }
+    return {"images": tf.cast(image, tf.float32), "bounding_boxes": bounding_boxes}
+
+
+def load_pascal_voc(split, dataset, bounding_box_format):
+    ds = tfds.load(dataset, split=split, with_info=False, shuffle_files=True)
+    ds = ds.map(
+        lambda x: unpackage_raw_tfds_inputs(x, bounding_box_format=bounding_box_format),
+        num_parallel_calls=tf.data.AUTOTUNE,
+    )
+    return ds
+
+
+train_ds = load_pascal_voc(
+    split="train", dataset="voc/2007", bounding_box_format="xywh"
+)
+eval_ds = load_pascal_voc(split="test", dataset="voc/2007", bounding_box_format="xywh")
+eval_ds = load_pascal_voc(dataset="voc/2007", split="test", bounding_box_format="xywh")
+
+train_ds = train_ds.shuffle(BATCH_SIZE * 4)
+visualization.visualize_dataset(train_ds, value_range=(0, 255), rows=2, cols=2)
+"""
+Next, lets batch our data.
+
+In KerasCV object detection tasks it is recommended that
```

Comment:
Why do the batches themselves need to be ragged? That doesn't seem to be related to image size of having a different number of labels in a dict.